OMEdit/OMEditLIB/Modeling/Model.cpp

```diff
@@ -1669,19 +1881,50 @@ namespace ModelInstance
     }
   }
 
-  QString Part::getName() const
+  QString Part::getName(bool includeSubscripts) const
   {
-    if (mSubScripts.isEmpty()) {
+    if (mSubScripts.isEmpty() || !includeSubscripts) {
       return mName;
     } else {
       return mName % "[" % mSubScripts.join(",") % "]";
     }
   }
 
+  Name::Name() = default;
+
+  Name::Name(const QString &str)
+  {
+    foreach (const QString &part, str.split('.')) {
```

Comment:
Fixed, but using `StringHandler::getFirstWordBeforeDot/removeFirstWordAfterDot` instead since `wordsBeforeAfterFirstDot` doesn't seem to be intended to be used directly. It's not exactly optimal, it would be better if we had some proper parsing instead, but I guess it works.

Also, isn't `removeFirstWordAfterDot` misnamed and should be `removeFirstWordBeforeDot`? That's what the comment for it says, and what it seems to be doing.